src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerManager.cs

```diff
@@ -224,56 +232,63 @@ private void ClearCompilationScopeMap(AnalyzerAndOptions analyzerAndOptions, Ana
                 }
             });
 
-            EventHandler<Exception> handler = null;
+            // Force evaluate and report exception diagnostics from LocalizableString.ToString().
             Action<Exception, DiagnosticAnalyzer, Diagnostic> onAnalyzerException = analyzerExecutor.OnAnalyzerException;
             if (onAnalyzerException != null)
             {
-                handler = new EventHandler<Exception>((sender, ex) =>
+                var handler = new EventHandler<Exception>((sender, ex) =>
                 {
                     var diagnostic = AnalyzerExecutor.CreateAnalyzerExceptionDiagnostic(analyzer, ex);
                     onAnalyzerException(ex, analyzer, diagnostic);
                 });
 
-                // Subscribe for exceptions from lazily evaluated localizable strings in the descriptors.
-                // REVIEW: find out better way to handle these exception handlers. right now, it can leak
-                //         so easily unless ClearAnalyzerState is called from host properly
                 foreach (var descriptor in supportedDiagnostics)
                 {
-                    descriptor.Title.OnException += handler;
-                    descriptor.MessageFormat.OnException += handler;
-                    descriptor.Description.OnException += handler;
+                    ForceLocalizableStringExceptions(descriptor.Title, handler);
+                    ForceLocalizableStringExceptions(descriptor.MessageFormat, handler);
+                    ForceLocalizableStringExceptions(descriptor.Description, handler);
                 }
             }
 
-            return Tuple.Create(supportedDiagnostics, handler);
+            return supportedDiagnostics;
+        }
+
+        private static void ForceLocalizableStringExceptions(LocalizableString localizableString, EventHandler<Exception> handler)
+        {
+            if (localizableString.CanThrowExceptions)
```

Comment:
why is this still here? so what happen if this ever ran? is it guaranteed that this never run in IDE or OOP?